### ISSUE TITLE
Metamask Mishandles ENS ERC-721 Transfer, Submits Wrong Token ID

### ISSUE THREAD

**Issue by kylebakerio** [MetaMask/metamask-mobile#4092]
**Describe the bug**
Attempting to transfer my ENS domain name (kyle3.eth) from one wallet to another, using Metamask Mobile's built-in ERC-721 transfer functionality, fails.
1. First, it just gives a cryptic "intrinsic gas too low" error. (This is almost certainly because the transaction being simulated is with faulty parameters, as we'll see later.)
2. Corrected the gas limit manually by looking up how much gas should be allocated for an ERC-721 transfer on etherscan (84904, rounded to 85000) (see [here](https://etherscan.io/gastracker)).
3. Then, transaction fails. ([Transaction here](https://etherscan.io/tx/0xb2da09b30f22714c04164c137ceda32723e83242d24db5b95929eccf8750ca08))
4. Looking up [transaction on tenderly](https://dashboard.tenderly.co/tx/mainnet/0xb2da09b30f22714c04164c137ceda32723e83242d24db5b95929eccf8750ca08/debugger?trace=0) shows that it fails at the check for the token to not be expired, within `ownerOf()`
```
    function ownerOf(uint256 tokenId) public view returns (address) {
        require(expiries[tokenId] > now);
        return super.ownerOf(tokenId);
    }
```
5. Looking at the state in that transaction within tenderly, I can see that the `from` and `to` arguments are correct, **but the `tokenId` argument appears to be wrong**. Thus, the failure is because the (presumably nonexistent) token does not have a valid expiration.

**Actual** Token ID [according to etherscan](https://etherscan.io/enslookup-search?search=kyle3.eth) is:
67389638965262593448462623680994207765963563118284578216830450145622015809311
(This also validates, if I look up the `ownerOf` for this one on [the ENS contract](https://etherscan.io/address/0x57f1887a8bf19b14fc0df6fd9b2acc9af147ea85#readContract) for it, I get my address)

Token ID in the transaction (therefore, injected/generated by metamask mobile app) according to tenderly is:
67389638965262590220318309333039877597547483020474590898904772124586079682560
(Again, validating by checking `ownerOf` reverts for this one)

Seems like a precision error, perhaps? They're both the same as of `6738963896526259` but then diverge after that number, while being the same length.

Looking on Etherscan to confirm, we see that argument 3 in the failed transaction is
94fd2f196cf1f800000000000000000000000000000000000000000000000000
which, when put through a hex->decimal converter, is:
67389638965262590220318309333039877597547483020474590898904772124586079682560
which matches what we see on tenderly

What is especially frustrating is I added my account to Metamask Mobile only because the chrome extension cannot handle NFT transfers but the docs inform that the mobile app can / should be used for that feature.

I'll just be doing it manually using etherscan and specifying the arguments manually, it seems.

Note: I'm sure the "intrinsic gas error" message is because Metamask Mobile predicted the transaction would fail. Better error messaging would be helpful.

**Screenshots**
N/A

**To Reproduce**
Should be clear enough from above; if not, let me know, will expand

**Expected behavior**
Should be clear

**Smartphone (please complete the following information):**
 - Device: OnePlus 9
 - OS: Oxygen OS, fully updated, Android 12, security update March 5, 2022
 - App Version: 4.3.1 (859)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

**Comment by Fatxx** [comment]
Check if it's related with https://github.com/MetaMask/metamask-mobile/pull/4636

**Comment by wdv4758h** [comment]
> Check if it's related with #4636

This seems to be related to the PR I sent. I was indeed having problem when doing NFT transfer due to Token ID is mis-converted.

I spent time digging the reason about MetaMask NFT transfer failure, while at the same time transfer via my other script are just fine. And then I found out the token id in transaction is wrong. So I search the code related to NFT transfer and find out a token id conversion issue if the token id is big.
